File: iOS/Advanced_Demo/Advanced/Common/BeautyCommon/VolcBeautyCommon/EffectResource.m
//
//  EffectResource.m
//  quickstart

#import "EffectResource.h"

@implementation EffectResource

#ifdef DEBUG



#define CVLicenseName @"rtc_test_20230524_20230831_vertc.veRTCDemo.ios_4.4.2_347.licbag"

#else

#define CVLicenseName @"rtc_test_20230524_20230831_com.ss.rtc.demo.advanced.ios_4.4.2_347.licbag"

#endif


+ (NSString *)licensePath {
    NSString *bunldePath = [[NSBundle mainBundle] pathForResource:@"LicenseBag" ofType:@"bundle"];
    NSString *licensePath = [NSString stringWithFormat:@"%@/%@", bunldePath, CVLicenseName];

    [self checkPathExsit:licensePath];

    return licensePath;
}

+ (NSString *)modelPath {
    NSString *modelPath = [[NSBundle mainBundle] pathForResource:@"ModelResource" ofType:@"bundle"];

    [self checkPathExsit:modelPath];

    return modelPath;
}

+ (NSString *)getByteEffectPortraitPath {
    NSString *portraitPrefix = [[NSBundle mainBundle] pathForResource:@"StickerResource" ofType:@"bundle"];
    NSString *name = @"matting_bg";
    NSString *path = [NSString stringWithFormat:@"%@/stickers/%@", portraitPrefix, name];
    [self checkPathExsit:path];

    return path;
}

+ (NSString *)beautyCameraPath {
    NSString *bunldePath = [[NSBundle mainBundle] pathForResource:@"ComposeMakeup" ofType:@"bundle"];
    NSString *beautyCameraPath = [NSString stringWithFormat:@"%@/ComposeMakeup/beauty_IOS_live", bunldePath];

    [self checkPathExsit:beautyCameraPath];

    return beautyCameraPath;
}

+ (NSString *)reshapeCameraPath {
    NSString *bunldePath = [[NSBundle mainBundle] pathForResource:@"ComposeMakeup" ofType:@"bundle"];
    NSString *reshapeCameraPath = [NSString stringWithFormat:@"%@/ComposeMakeup/reshape_live", bunldePath];

    [self checkPathExsit:reshapeCameraPath];

    return reshapeCameraPath;
}

+ (NSString *)stickerPathWithName:(NSString *)stickerName {
    NSString *bunldePath = [[NSBundle mainBundle] pathForResource:@"StickerResource" ofType:@"bundle"];
    NSString *stickerPath = [NSString stringWithFormat:@"%@/stickers/%@", bunldePath, stickerName];

    [self checkPathExsit:stickerPath];

    return stickerPath;
}

+ (NSString *)filterPathWithName:(NSString *)filterName {
    NSString *bunldePath = [[NSBundle mainBundle] pathForResource:@"FilterResource" ofType:@"bundle"];
    NSString *filterPath = [NSString stringWithFormat:@"%@/Filter/%@", bunldePath, filterName];

    [self checkPathExsit:filterPath];

    return filterPath;
}

+ (void)checkPathExsit:(NSString *)path {
    BOOL isDirectory;
    BOOL exist = [[NSFileManager defaultManager] fileExistsAtPath:path isDirectory:&isDirectory];

    if (!exist) {
        NSLog(@"EffectResource path is not exist \n%@", path);
    }
}
@end

File: iOS/ApiExample/ApiExample/ImportantComponents/Beauty/VolcBeauty/EffectResource.swift
//
//  EffectResource.swift
//  ApiExample
//
//  Created by bytedance on 2023/11/2.
//  Copyright © 2021 bytedance. All rights reserved.
//

import Foundation

class EffectResource {
    
    static func licensePath() -> String {
        let bundlePath = Bundle.main.path(forResource: "LicenseBag", ofType: "bundle") ?? ""
        let licensePath = "\(bundlePath)/\(CVLicenseName)"
        checkPathExist(path: licensePath)
        
        return licensePath
    }
    
    static func modelPath() -> String {
        let modelPath = Bundle.main.path(forResource: "ModelResource", ofType: "bundle") ?? ""
        checkPathExist(path: modelPath)
        
        return modelPath
    }
    
    static func getByteEffectPortraitPath() -> String {
        let portraitPrefix = Bundle.main.path(forResource: "StickerResource", ofType: "bundle") ?? ""
        let name = "matting_bg"
        let path = "\(portraitPrefix)/stickers/\(name)"
        checkPathExist(path: path)
        
        return path
    }
    
    static func beautyCameraPath() -> String {
        let bundlePath = Bundle.main.path(forResource: "ComposeMakeup", ofType: "bundle") ?? ""
        let beautyCameraPath = "\(bundlePath)/ComposeMakeup/beauty_IOS_lite"
        checkPathExist(path: beautyCameraPath)
        
        return beautyCameraPath
    }
    
    static func reshapeCameraPath() -> String {
        let bundlePath = Bundle.main.path(forResource: "ComposeMakeup", ofType: "bundle") ?? ""
        let reshapeCameraPath = "\(bundlePath)/ComposeMakeup/reshape_lite"
        checkPathExist(path: reshapeCameraPath)
        
        return reshapeCameraPath
    }
    
    static func stickerPath(withName stickerName: String) -> String {
        let bundlePath = Bundle.main.path(forResource: "StickerResource", ofType: "bundle") ?? ""
        let stickerPath = "\(bundlePath)/stickers/\(stickerName)"
        checkPathExist(path: stickerPath)
        
        return stickerPath
    }
    
    static func filterPath(withName filterName: String) -> String {
        let bundlePath = Bundle.main.path(forResource: "FilterResource", ofType: "bundle") ?? ""
        let filterPath = "\(bundlePath)/Filter/\(filterName)"
        checkPathExist(path: filterPath)
        
        return filterPath
    }
    
    static func checkPathExist(path: String) {
        let exist = FileManager.default.fileExists(atPath: path)
        
        if !exist {
            NSLog("EffectResource path does not exist: \n\(path)")
            ToastComponents.shared.show(withMessage: "资源不存在:\(path)")
        }
    }
}
